modules/seqimporter/src/main/scala/gem/seqimporter/SequenceImporter.scala

```diff
@@ -0,0 +1,152 @@
+    * sequence XML.
+    */
+  private def fetchSequenceXml(oid: Observation.Id): Task[Elem] = {
+    // WDBA implements an XML-RPC service for fetching sequences.  We post the
+    // following XML and it returns the matching sequence.
+    val req: Task[Request] =
+      Request(Method.POST, Uri.unsafeFromString("http://gsodb.gemini.edu:8442/wdba")).withBody {
+        <methodCall>
+          <methodName>WDBA_Exec.getSequence</methodName>
+          <params><param><value><string>{oid}</string></value></param></params>
+        </methodCall>
+      }
+
+    // Run the request on the ODB, which returns the sequence XML if all goes
+    // well wrapped in XML-RPC elements.
+    client.expect[Elem](req).map { elem =>
+      // Fish the return value out of the XML-RPC result.
+      val xmlTxt = (elem \\ "value").text
+
+      // Drop the preliminary cruft and then parse the sequence XML.  We're
+      // assuming that the observation was found in the ODB and all went well
+      // or else this will fail with an exception.  We could fish out XML-RPC
+      // failure results and return more accurate failure information if needed.
+      XML.loadString(xmlTxt.lines.drop(3).mkString("\n"))
+    }
+  }
+
+  /** Converts the OCS2 sequence XML into the new sequence model.  Each step
+    * includes the instrument configuration and its completion state.
+    */
+  private def parseSequenceXml(oid: Observation.Id, xml: Elem): List[Step[(InstrumentConfig, Boolean)]] = {
+    // Extracts the steps in the XML sequence to a simple list of Maps where
+    // each Map is all the keys and values that apply to the step.
+    def configMaps(xml: Elem): List[ConfigMap] = {
+      def extractSystem(configMap: ConfigMap, sys: Node): ConfigMap = {
+        val sysName = (sys \ "@name").text
+
+        (configMap /: (sys \ "param")) { (map, param) =>
+          val paramName  = (param \ "@name").text
+          val paramValue = (param \ "@value").text
+          map + (s"$sysName:$paramName" -> paramValue)
+        }
+      }
+
+      def extractStep(configMap: ConfigMap, step: Node): ConfigMap =
+        (configMap /: (step \ "system")) { extractSystem }
+
+      (xml \ "step").toList.scanLeft(EmptyConfigMap) { extractStep }.tail
+    }
+
+    configMaps(xml).map(ConfigReader.unsafeStep)
+  }
+
+  /** Fetches a Sequence from the ODB using its WDBA service, translating it
+    * into the new sequence model in memory.
+    */
+  def fetchSequence(oid: Observation.Id): Task[List[Step[(InstrumentConfig, Boolean)]]] =
+    fetchSequenceXml(oid).map { parseSequenceXml(oid, _) }
+
+  private val configureLogging: Task[Unit] =
+    Task.delay(List(
+      "edu.gemini.spModel.type.SpTypeUtil"
+    ).map(Logger.getLogger).foreach(_.setLevel(Level.OFF)))
+
+  // We need a minimal program/observation in order to insert steps.  The
+  // seqexec doesn't actually use these but they have to be there to satisfy
```

Comment:
I don't think the seqexec will need to read the program or observation, though if the information were correct it could certainly provide more information like the observation title.  These need to exist in the database though because there are constraints such as the one that says each step's `observation_id` references the `observation_id` column of the `observation` table.